data-transfer/src/main/java/it/eng/datatransfer/rest/api/RestArtifactController.java

```diff
@@ -36,26 +40,21 @@ public RestArtifactController(RestArtifactService restArtifactService) {
 	 * @return
 	 */
     @RequestMapping(path = "/{transactionId}", method = { RequestMethod.GET, RequestMethod.POST })
-    protected ResponseEntity<String> getArtifact(HttpServletResponse response,
+    protected void getArtifact(HttpServletResponse response,
     												@RequestHeader(required = false) String authorization,
 										    		@PathVariable String transactionId,                                       
 										    		@RequestBody(required = false) JsonNode jsonBody) {
     
-		String rr = restArtifactService.getArtifact(transactionId, jsonBody);
-		return ResponseEntity.ok()
-                .contentType(MediaType.APPLICATION_JSON)
-                .body(rr);
-		/*
-		 * Uncomment this part when TransferProcess.State.INITIAL is done
-		 * this logic requires to have fileId present in TransferPRocess
-		try (ServletOutputStream outputStream = response.getOutputStream()) {
-			GridFsResource attachment = restArtifactService.streamAttachment(transactionId);
+    	log.info("Starting data download");
+		
+		GridFsResource attachment = restArtifactService.streamAttachment(transactionId);
+		try {
+			response.setStatus(HttpStatus.OK.value());
 			response.setHeader("Content-Disposition", "attachment;filename=\"" + attachment.getFilename() + "\"");
 			response.addHeader("Content-type", attachment.getContentType());
```

Comment:
What happens if you upload something without content-type? Should we create some fail safe use-case, if no content-type is present either remove it from headers or put plain/text

Reply:
![image](https://github.com/user-attachments/assets/93871bb2-ca30-426a-ab18-136c0f84131d)
![image](https://github.com/user-attachments/assets/a37fc194-680f-4a56-98e2-9819a37a123c)

This is the part were we upload the file, and by default the Content-type is nullable, sadly. I would there rather make the CT mandatory on upload then to check it on download and have to go back to uploading again. Or even better on both sides. Do you agree?